src/status_im/components/tabs/views.cljs

```diff
@@ -1,27 +1,33 @@
 (ns status-im.components.tabs.views
-  (:require-macros [status-im.utils.views :refer [defview letsubs]])
-  (:require [re-frame.core :refer [dispatch]]
+  (:require-macros [status-im.utils.views :refer [defview]]
+                   [cljs.core.async.macros :as am])
+  (:require [cljs.core.async :as async]
+            [reagent.core :as r]
+            [re-frame.core :refer [subscribe dispatch]]
             [status-im.components.react :as react]
             [status-im.components.icons.vector-icons :as vi]
-            [status-im.components.tabs.styles :as styles]))
+            [status-im.components.tabs.styles :as styles]
+            [status-im.utils.platform :as p]))
 
-(defn- tab [{:keys [view-id title icon-active icon-inactive selected-view-id]}]
+(defn- tab [{:keys [view-id title icon-active icon-inactive selected-view-id on-press]}]
   (let [active? (= view-id selected-view-id)]
     [react/touchable-highlight {:style    styles/tab
                                 :disabled active?
-                                :on-press  #(dispatch [:navigate-to-tab view-id])}
+                                :on-press  #(if on-press (do (println "tab click" view-id) (on-press view-id)) (dispatch [:navigate-to-tab view-id]))}
```

Comment:
is it intentional or debug leftover ?

Reply:
Intentional, `tab` component is not fully abstract now (this behavior is needed for `main-tabs`)

Reply:
Oops I missed the println ..